impeller/entity/contents/text_contents.cc

```diff
@@ -224,4 +224,17 @@ bool TextContents::Render(const ContentContext& renderer,
                       cmd);
 }
 
+// |Contents|
+void TextContents::AdoptLazyGlyphAtlas(
```

Comment:
It doesn't look like anything here is actually reassigning the `lazy_atlas_`, and so when the picture text renders, it's still going to be using the atlas from the picture canvas.

Since we shouldn't perform the merge here, we could get away with just making the existing glyph atlas setter virtual instead.